transcom/mymove
# B-20758 TPPS Paid Payment Request Status Update INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20758)

## Summary

When we process the TPPS Paid Invoice Report, we also want to update the status of all the payment requests in the report itself to have a status of PAID.

Each row in the TPPS Paid Invoice Report refers to one payment service item - this means we can have many rows that all refer to the same payment request.

Given we only need to update the payment request status once, the logic for updating the payment request status keeps track of which payment requests that have already updated with a `map`. For each row of the TPPS report, we find the corresponding payment request - then we check if that payment request is in our map already. If it is, that means we already updated the status, so we don't need to do it again.

Work in this PR:
- added logic for updating payment request status to PAID
- added logging to give specific information that mentions the payment request number, payment request status, and TPPS data invoice number that was updated (_payment request number matches invoice number_)
- added a new test specifically for checking payment request status in pkg/services/invoice/process_tpps_paid_invoice_report_test.go
- also cleaned up the other tests in that file (didn't need to be calling `factory.BuildPaymentRequestToInterchangeControlNumber` for any of the test payment requests, was initially a copy/paste from other EDI processing tests) 

Note:
The second AC mentioning seeing the "TPPS PAID" displayed next to the payment request number on the UI was already covered/met by [B-20674](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20674)

### How to test

1. This is another of those "look really closely at the tests" PRs. Feel free to debug and check it out though!
2. Here are some screenshots to show what we are doing with the map logic
3. Here is what a parsed TPPS report looks like - note the _duplicate_ InvoiceNumbers on the right. The invoice_number for each TPPS report is how we find the matching payment request - it will match the payment request's payment_request_number.

![image](https://github.com/user-attachments/assets/c66c748d-2608-4c6e-98ad-84cf791b9658)

4. Here is the map of the payment request with key:value pairs of id:payment_request_number. These show that we only did 4 transactions to update the payment request status once for each payment request in the TPPS report - more efficient than updating the payment request status when we don't need to.

![image](https://github.com/user-attachments/assets/1d4eb802-492a-4bfd-a04b-096e6afecb65)
